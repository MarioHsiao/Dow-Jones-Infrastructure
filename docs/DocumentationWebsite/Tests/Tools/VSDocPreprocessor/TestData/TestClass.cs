using System;

namespace DowJones.Documentation.Tests.Tools.VSDocPreprocessor.TestData
{
    /// <summary>
    /// A test enum
    /// </summary>
    /// <remarks>
    /// This class is awesome.
    /// </remarks>
    /// <example>
    /// To use this class, just instantiate it like this:
    /// <code>var class = new TestClass("Awesome!");</code>
    /// </example>
    public enum TestEnum
    {
        /// <summary>
        /// The first value
        /// </summary>
        Value1,

        /// <summary>
        /// The second value
        /// </summary>
        Value2
    }

    /// <summary>
    /// A test class with a name that looks very similar to TestClass
    /// </summary>
    public class TestClassBuilder
    {
        /// <summary>
        /// Creates a new TestClassBuilder
        /// </summary>
        public TestClassBuilder()
        {
        }

        /// <summary>
        /// Builds the TestClass
        /// </summary>
        /// <param name="class">The TestClass to build</param>
        public void Build(TestClass @class)
        {
        }
    }

    /// <summary>
    /// A test class that emits VSDoc documentation
    /// </summary>
    public class TestClass
    {
        /// <summary>
        /// A read/write static property
        /// </summary>
        public static string StaticProperty { get; set; }

        /// <summary>
        /// A test enum declared as an inner class
        /// </summary>
        public enum TestInnerEnum
        {
            /// <summary>
            /// The first value
            /// </summary>
            Value1,

            /// <summary>
            /// The second value
            /// </summary>
            Value2
        }

        /// <summary>
        /// Public event
        /// </summary>
        public event EventHandler Event;

        /// <summary>
        /// A primary type property
        /// </summary>
        public string Property1 { get; set; }

        /// <summary>
        /// This is a complex type property
        /// </summary>
        public Tuple<int, string> Property2 { get; set; }

        /// <summary>
        /// The default constructor
        /// </summary>
        public TestClass()
        {
        }

        /// <summary>
        /// A constructor with one parameter
        /// </summary>
        /// <param name="param1">The first parameter</param>
        public TestClass(string param1)
        {
        }

        /// <summary>
        /// A constructor with two parameters
        /// </summary>
        /// <param name="param1">The first parameter</param>
        /// <param name="param2">The second parameter</param>
        public TestClass(string param1, string param2)
        {
        }


        /// <summary>
        /// A method that doesn't actually do or return anything
        /// </summary>
        /// <param name="param">The parameter for the method</param>
        public void VoidMethod(int param)
        {
        }

        /// <summary>
        /// A method that doesn't accept parameters
        /// </summary>
        public void MethodWithNoParameters()
        {
        }

        /// <summary>
        /// A method that returns a complex type
        /// </summary>
        /// <param name="param">The parameter for the method</param>
        public Uri ComplexMethod(Uri param)
        {
            return param;
        }


        /// <summary>
        /// The static version of the VoidMethod
        /// </summary>
        /// <param name="param">The parameter for the method</param>
        public static void StaticVoidMethod(string param)
        {
        }
    }
}
