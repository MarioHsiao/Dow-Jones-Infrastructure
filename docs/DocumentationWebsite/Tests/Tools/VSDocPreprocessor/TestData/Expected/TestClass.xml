<?xml version="1.0" encoding="utf-8"?>
<type assembly="DowJones.Documentation.Tests" fullName="DowJones.Documentation.Tests.Tools.VSDocPreprocessor.TestData.TestClass" name="TestClass" namespace="DowJones.Documentation.Tests.Tools.VSDocPreprocessor.TestData">
  <summary>A test class that emits VSDoc documentation</summary>
  <constructors>
    <constructor name="TestClass()">
      <summary>The default constructor</summary>
      <parameters />
    </constructor>
    <constructor name="TestClass(string)">
      <summary>A constructor with one parameter</summary>
      <parameters>
        <parameter name="param1" type="string">
          <remarks>The first parameter</remarks>
        </parameter>
      </parameters>
    </constructor>
    <constructor name="TestClass(string, string)">
      <summary>A constructor with two parameters</summary>
      <parameters>
        <parameter name="param1" type="string">
          <remarks>The first parameter</remarks>
        </parameter>
        <parameter name="param2" type="string">
          <remarks>The second parameter</remarks>
        </parameter>
      </parameters>
    </constructor>
  </constructors>
  <events>
    <event name="Event">
      <summary>Public event</summary>
    </event>
  </events>
  <methods>
    <method name="ComplexMethod(Uri)">
      <summary>A method that returns a complex type</summary>
      <parameters>
        <parameter name="param" type="Uri">
          <remarks>The parameter for the method</remarks>
        </parameter>
      </parameters>
    </method>
    <method name="MethodWithNoParameters()">
      <summary>A method that doesn't accept parameters</summary>
      <parameters />
    </method>
    <method name="StaticVoidMethod(string)">
      <summary>The static version of the VoidMethod</summary>
      <parameters>
        <parameter name="param" type="string">
          <remarks>The parameter for the method</remarks>
        </parameter>
      </parameters>
    </method>
    <method name="VoidMethod(int)">
      <summary>A method that doesn't actually do or return anything</summary>
      <parameters>
        <parameter name="param" type="int">
          <remarks>The parameter for the method</remarks>
        </parameter>
      </parameters>
    </method>
  </methods>
  <properties>
    <property name="Property1">
      <summary>A primary type property</summary>
    </property>
    <property name="Property2">
      <summary>This is a complex type property</summary>
    </property>
    <property name="StaticProperty">
      <summary>A read/write static property</summary>
    </property>
  </properties>
</type>